firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -761,6 +762,23 @@ class BrowserCoordinator: BaseCoordinator,
             performIfCoordinatorRootVCIsPresented(SettingsCoordinator.self) {
                 didFinishSettings(from: $0)
             }
+        case .syncMenuOpened:
+            guard uuid != windowUUID else { return }
```

Comment:
nit: can we add this at the top level since it seems all cases repeat this code?

Reply:
My thinking is that it would be safer to keep these checks within the individual cases, the reason being that these are general-purpose events and not all future cases handled by this switch will necessarily have that same requirement (which means if someone adds a new case and doesn't notice the guard at the top it may not work correctly). Having said that I totally understand the rationale for DRYing this out so if you have strong feelings about it I'm happy to change it, just LMK.

Reply:
thanks for the rationale! lets keep as-is!